realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -229,9 +230,11 @@ public void logout() {
 
             // invalidate all pending refresh_token queries
             for (SyncConfiguration syncConfiguration : realms.keySet()) {
-                SyncSession session = SyncManager.getSession(syncConfiguration);
-                if (session != null) {
+                try {
+                    SyncSession session = SyncManager.getSession(syncConfiguration);
                     session.clearScheduledAccessTokenRefresh();
+                } catch (RealmException e) {
+                    // No session, either the Realm was not opened or session was removed.
```

Comment:
RealmException is kinda generic. It would feel safer if we did more checking here, like checking the error message before just throwing the error away?